processor/src/chiplets/aux_trace/mod.rs

```diff
@@ -245,3 +304,1067 @@ impl AuxColumnBuilder<ChipletsVTableUpdate, ChipletsVTableRow, u32> for Chiplets
+
+    let op_label = RETURN_HASH_LABEL;
+    let op_label = if addr_to_hash_cycle(helper_0 + s4.mul_small(8) - ONE) == 0 {
+        op_label + 16
+    } else {
+        op_label + 32
+    };
+
+    let sum_output = alphas[8..12]
+        .iter()
+        .rev()
+        .enumerate()
+        .fold(E::ZERO, |acc, (i, x)| acc + x.mul_base(s6_s9[i]));
+    let v_output_old = alphas[0]
+        + alphas[1].mul_base(Felt::from(op_label))
+        + alphas[2].mul_base(helper_0 + s4.mul_small(8) - ONE)
+        + sum_output;
+
+    let op_label = MR_UPDATE_NEW_LABEL;
+    let op_label = if addr_to_hash_cycle(helper_0 + s4.mul_small(8)) == 0 {
+        op_label + 16
+    } else {
+        op_label + 32
+    };
+    let sum_input = alphas[8..12]
+        .iter()
+        .rev()
+        .enumerate()
+        .fold(E::ZERO, |acc, (i, x)| acc + x.mul_base(s10_s13[i]));
+    let v_input_new = alphas[0]
+        + alphas[1].mul_base(Felt::from(op_label))
+        + alphas[2].mul_base(helper_0 + s4.mul_small(8))
+        + alphas[3].mul_base(s5)
+        + sum_input;
+
+    let op_label = RETURN_HASH_LABEL;
+    let op_label = if addr_to_hash_cycle(helper_0 + s4.mul_small(16) - ONE) == 0 {
+        op_label + 16
+    } else {
+        op_label + 32
+    };
+
+    let sum_output = alphas[8..12]
+        .iter()
+        .rev()
+        .enumerate()
+        .fold(E::ZERO, |acc, (i, x)| acc + x.mul_base(s0_s3_nxt[i]));
+    let v_output_new = alphas[0]
+        + alphas[1].mul_base(Felt::from(op_label))
+        + alphas[2].mul_base(helper_0 + s4.mul_small(16) - ONE)
+        + sum_output;
+
+    v_input_new * v_input_old * v_output_new * v_output_old
+}
+
+// CHIPLETS RESPONSES
+// ================================================================================================
+
+/// Constructs the responses from the chiplets to the other VM-components at row i.
+fn chiplets_responses<E>(main_trace: &MainTrace, alphas: &[E], i: usize) -> E
+where
+    E: FieldElement<BaseField = Felt>,
+{
+    let selector0 = main_trace.chiplet_selector_0(i);
+    let selector1 = main_trace.chiplet_selector_1(i);
+    let selector2 = main_trace.chiplet_selector_2(i);
+    let selector3 = main_trace.chiplet_selector_3(i);
+    let selector4 = main_trace.chiplet_selector_4(i);
+
+    if selector0 == ZERO {
+        return build_hasher_chiplet_responses(
+            main_trace, i, alphas, selector1, selector2, selector3,
+        );
+    }
+
+    if selector0 == ONE && selector1 == ZERO {
+        return build_bitwise_chiplet_responses(main_trace, i, selector2, alphas);
+    }
+
+    if selector0 == ONE && selector1 == ONE && selector2 == ZERO {
+        return build_memory_chiplet_responses(main_trace, i, selector3, alphas);
+    }
+
+    if selector0 == ONE
+        && selector1 == ONE
+        && selector2 == ONE
+        && selector3 == ZERO
+        && selector4 == ONE
+    {
+        build_kernel_chiplet_responses(main_trace, i, alphas)
+    } else {
+        E::ONE
+    }
```

Comment:
nit: we can probably write this as a sequence of `else if` statements like so:
```Rust
if selector0 == ZERO {
    build_hasher_chiplet_responses(main_trace, i, alphas, selector1, selector2, selector3)
} else if selector1 == ZERO {
    debug_assert_eq!(selector0, ONE);
    build_bitwise_chiplet_responses(main_trace, i, selector2, alphas)
} else if selector2 == ZERO {
    debug_assert_eq!(selector0, ONE);
    debug_assert_eq!(selector1, ONE);
    build_memory_chiplet_responses(main_trace, i, selector3, alphas)
} else if selector3 == ZERO {
    debug_assert_eq!(selector0, ONE);
    debug_assert_eq!(selector1, ONE);
    debug_assert_eq!(selector2, ONE);
    build_kernel_chiplet_responses(main_trace, i, alphas)
} else {
    debug_assert_eq!(selector0, ONE);
    debug_assert_eq!(selector1, ONE);
    debug_assert_eq!(selector2, ONE);
    debug_assert_eq!(selector3, ONE);
    E::ONE
}
```

Reply:
Makes sense. Only that I have add a check for `selector4 == ONE` before calling `build_kernel_chiplet_responses`